packages/vizbuilder/src/helpers/chartconfig.js

```diff
@@ -0,0 +1,330 @@
+
+    return config;
+  },
+  lineplot(commonConfig, query) {
+    const {timeDrilldown, drilldown, measure, moe} = query;
+
+    const drilldownName = timeDrilldown.name;
+    const measureName = measure.name;
+
+    const config = {
+      ...commonConfig,
+      discrete: "x",
+      groupBy: drilldown.hierarchy.levels[1].name,
+      x: drilldownName,
+      xConfig: {title: drilldownName},
+      y: measureName,
+      yConfig: {title: measureName}
+    };
+
+    if (moe) {
+      const moeName = moe.name;
+
+      config.confidence = [
+        d => d[measureName] - d[moeName],
+        d => d[measureName] + d[moeName]
+      ];
+      config.confidenceConfig = {
+        fillOpacity: 0.15
+      };
+    }
+
+    return config;
+  },
+  stacked(commonConfig, query) {
+    const config = this.lineplot(commonConfig, query);
+    return config;
+  },
+  treemap(commonConfig, query) {
+    const {drilldown, measure} = query;
+
+    const drilldownName = drilldown.name;
+    const measureName = measure.name;
+
+    const config = {
+      ...commonConfig,
+      groupBy: drilldownName,
+      y: measureName,
+      yConfig: {title: measureName}
+    };
+
+    return config;
+  }
+};
+
+export default function createChartConfig({
+  activeType,
+  query,
+  availableKeys,
+  year
+}) {
+  const availableCharts = new Set(
+    activeType ? [activeType] : Object.keys(charts)
+  );
+
+  const measureName = query.measure.name;
+  const getMeasureName = d => d[measureName];
+
+  const commonConfig = {
+    height: activeType ? 500 : 400,
+    legend: false,
+
+    tooltip: true,
+    tooltipConfig: tooltipConfig(query),
+
+    duration: 0,
+    loadingMessage: "Loading",
+    sum: getMeasureName,
+    value: getMeasureName,
+    totalConfig: {
+      fontSize: 14
+    }
+  };
+
+  const colorScale = {
+    colorScale: measureName,
+    colorScalePosition: "bottom",
+    colorScaleConfig: {width: 0, height: 0}
+  };
+
+  if (!activeType) {
+    const hasTimeDim = availableKeys.has("Year");
+    const hasGeoDim = availableKeys.has("State") || availableKeys.has("County");
+    const aggregatorType =
+      query.measure.annotations.aggregation_method ||
+      query.measure.aggregatorType ||
+      "UNKNOWN";
+
+    if (!hasTimeDim || year !== ALL_YEARS) {
+      availableCharts.delete("stacked");
+      availableCharts.delete("barchartyear");
+      availableCharts.delete("lineplot");
+    }
+    if (!hasGeoDim || year === ALL_YEARS) {
+      availableCharts.delete("geomap");
+    }
+    if (aggregatorType === "AVERAGE") {
+      availableCharts.delete("donut");
+      availableCharts.delete("stacked");
+      availableCharts.delete("treemap");
+      availableCharts.delete("histogram");
+    }
+    if (aggregatorType !== "SUM") {
+      availableCharts.delete("treemap");
+      availableCharts.delete("barchartyear");
+    }
+  }
+  else {
+    if (year !== ALL_YEARS) {
+      commonConfig.total = getMeasureName;
+      commonConfig.totalConfig = {
```

Comment:
remove this totalConfig, as it is styling that should be the user's responsibility.

Reply:
done in 096b8f5a0900b84d4341cca21e35db479343399a